server/src/quiz/quiz.service.ts

```
@@ -0,0 +1,277 @@
+   * @returns Random question with shuffled answers
+   */
+  async getRandomQuestion(
+    challengeId: string,
+    userId: string,
+  ): Promise<QuizQuestionDto> {
+    const availableQuestions = await this.prisma.quizQuestion.findMany({
+      where: {
+        challengeId,
+        userAnswers: {
+          none: { userId },
+        },
+      },
+      include: {
+        answers: true,
+      },
+    });
+
+    if (availableQuestions.length === 0) {
+      throw new NotFoundException('No available questions for this challenge');
+    }
+
+    const randomIndex = Math.floor(Math.random() * availableQuestions.length);
+    const selectedQuestion = availableQuestions[randomIndex];
+
+    const shuffledAnswers = this.shuffleArray<{
+      id: string;
+      answerText: string;
+      isCorrect?: boolean;
+    }>(selectedQuestion.answers);
+
+    return {
+      id: selectedQuestion.id,
+      questionText: selectedQuestion.questionText,
+      answers: shuffledAnswers.map(answer => ({
+        id: answer.id,
+        answerText: answer.answerText,
+      })),
+      pointValue: selectedQuestion.pointValue,
+      challengeId: selectedQuestion.challengeId,
+    };
+  }
+
+  /**
+   * Get a different question (for shuffle functionality)
+   * @param challengeId - Challenge ID
+   * @param userId - User ID
+   * @param excludeQuestionId - Question to exclude from selection
+   * @returns Different random question
+   */
+  async shuffleQuestion(
+    challengeId: string,
+    userId: string,
+    excludeQuestionId?: string,
+  ): Promise<QuizQuestionDto> {
+    const availableQuestions = await this.prisma.quizQuestion.findMany({
+      where: {
+        challengeId,
+        id: excludeQuestionId ? { not: excludeQuestionId } : undefined,
+        userAnswers: {
+          none: { userId },
+        },
+      },
+      include: {
+        answers: true,
+      },
+    });
+
+    if (availableQuestions.length === 0) {
```

This should throw an exception, since you have no questions left to give. Above in getRandomQuestion, you do this: 
if (availableQuestions.length === 0) {
      throw new NotFoundException('No available questions for this challenge');
    }